### PR TITLE
feat(lib): move job_engine back into nuvla package

### PR DESCRIPTION
**Pull request by ignacio-penas**
Closes https://github.com/SixSq/tasklist/issues/2991

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -4,5 +4,13 @@ skipsdist = True
...
-commands = pytest --junitxml=test-report.xml --cov=job_engine --cov-report=xml --cov-config=.coveragerc --cov-branch {posargs}
+allowlist_externals =
+    bash
+    pip
+
+commands =
+    pip install poetry
+    poetry build -f wheel
+    bash -c "pip install dist/nuvla_job_engine-$(poetry version -s)-py3-none-any.whl --force-reinstall --no-deps"
+    pytest --junitxml=test-report.xml --cov=nuvla --cov-report=xml --cov-config=.coveragerc --cov-branch {posargs}
```
[the remark this answers is not in the document]
Comment (ignacio-penas) [reply tox.ini:13]: Done. Thanks for the suggestion. Improves the workflow by a huge margin

File: tox.ini [hunk 2]
```diff
@@ -4,5 +4,10 @@ skipsdist = True
 
 [testenv]
 deps = -rrequirements.test.txt
-commands = pytest --junitxml=test-report.xml --cov=job_engine --cov-report=xml --cov-config=.coveragerc --cov-branch {posargs}
+allowlist_externals =
+    bash
+
+commands =
+    bash -c "ln -sf $(pwd)/nuvla/job_engine {envdir}/lib/python3.11/site-packages/nuvla/"
+    pytest --junitxml=test-report.xml --cov=nuvla --cov-report=xml --cov-config=.coveragerc --cov-branch {posargs}
```
Comment (konstan) [inline tox.ini:13]: @ignacio-penas Here is another improvement I'm suggesting. Which allows dynamic determination of the Python version as well.

```
bash -c "ln -sf $(pwd)/nuvla/job_engine {envdir}/lib/python{py_dot_ver}/site-packages/nuvla/"
```

**Review by konstan — CHANGES_REQUESTED** [review]
@ignacio-penas Please check comment with another improvement.

**Comment by konstan** [comment]
@ignacio-penas LGTM now.

File: tox.ini (hunk 2 above)
Comment (konstan) [reply tox.ini:13]: @ignacio-penas Thanks for incorporating the suggestion. I'm resolving this conversation.

**Review by konstan — APPROVED** [review]

**Pull request merged**